agenta-web/cypress/e2e/ai-critic-evaluation.cy.ts

```diff
@@ -0,0 +1,251 @@
+            cy.url().should("include", "/evaluations")
+        })
+    })
+
+    context("When you select evaluation in the absence of an API key", () => {
+        beforeEach(() => {
+            cy.visit("/apps")
+            cy.clickLinkAndWait('[data-cy="app-card-link"]')
+            cy.clickLinkAndWait('[data-cy="app-evaluations-link"]')
+        })
+
+        it("Should display modal", () => {
+            cy.get('[data-cy="evaluation-error-modal"]').should("not.exist")
+            cy.get('[data-cy="ai-critic-button"]').click()
+
+            cy.request({
+                url: `${Cypress.env().baseApiURL}/app_variant/list_apps/`,
+                method: "GET",
+            }).then((res) => {
+                expect(res.status).to.equal(200)
+                cy.request({
+                    url: `${Cypress.env().baseApiURL}/app_variant/list_variants/?app_name=${
+                        res.body[0].app_name
+                    }`,
+                    method: "GET",
+                }).then((response) => {
+                    expect(response.status).to.equal(200)
+                    cy.get('[data-cy="variants-dropdown"]')
+                        .find(".ant-dropdown-trigger")
+                        .each((dropdown, index) => {
+                            cy.wrap(dropdown).click()
+                            cy.get(".ant-dropdown-menu-item")
+                                .contains(response.body[index].variant_name)
+                                .click()
+                        })
+                })
+            })
+            cy.request({
+                url: `${Cypress.env().baseApiURL}/app_variant/list_apps/`,
+                method: "GET",
+            }).then((res) => {
+                expect(res.status).to.equal(200)
+                cy.request({
+                    url: `${Cypress.env().baseApiURL}/testsets/?app_name=${res.body[0].app_name}`,
+                    method: "GET",
+                }).then((response) => {
+                    expect(response.status).to.equal(200)
+                    cy.get('[data-cy="selected-testset"]')
+                        .find(".ant-dropdown-trigger")
+                        .each((dropdown, index) => {
+                            cy.wrap(dropdown).click()
+                            cy.get(".ant-dropdown-menu-item")
+                                .contains(response.body[index].name)
+                                .click()
+                        })
+                })
+            })
+
+            cy.clickLinkAndWait('[data-cy="start-new-evaluation-button"]')
+            cy.get('[data-cy="evaluation-error-modal"]').should("exist")
+            cy.get('[data-cy="evaluation-error-modal-ok-button"]').click()
+        })
+
+        it("Should display modal and naviagte to apikeys", () => {
+            cy.get('[data-cy="evaluation-error-modal"]').should("not.exist")
+            cy.get('[data-cy="ai-critic-button"]').click()
+
+            cy.request({
```

Comment:
@bekossy there are a lot of promise chaining in this due to requests. I wonder why you have to send api requests like this. Ideally, the cypress tests should only interact with the UI and should not be aware of the underlying API requests, let alone make requests themselves.